src/app/application-frame/application-frame.component.ts

```diff
@@ -29,12 +31,14 @@ export class ApplicationFrameComponent implements OnInit {
 
   public constructor(
     private readonly userTelemetryOrchestrationService: UserTelemetryOrchestrationService,
-    private readonly timeRangeService: TimeRangeService
+    private readonly timeRangeService: TimeRangeService,
+    private readonly addUserService: AddUserService
   ) {
     this.timeRangeHasInit$ = this.timeRangeService.getTimeRangeAndChanges().pipe(take(1));
   }
 
   public ngOnInit(): void {
+    this.addUserService.addUser().subscribe();
```

Comment:
Why call this method here and not in the `UserInfoService`?

As explained in another comment, the `addUser()` method will be added to the `UserPreferenceService`, which uses `UserInfoService` in its constructor.

To call the `this.userPreferenceService.addUser()` method inside `load()` of `UserInfoService`, we'll need to add `UserPreferenceService` inside the constructor of `UserInfoService`. 

Not sure if that'll work, having 2 services that both use each other's methods, scenario explained in below code snippet.

```js
class UserInfoService {
  constructor(private readonly userPreferenceService) {}
  
  load() {
    this.userPreferenceService.addUser();
    }
}

class UserPreferenceService {
  constructor(private readonly userInfoService: UserInfoService) {
    // user email is required by all methods in this service
  }
}
```

Reply:
I think calling it that way is required. Can't we have a method in `UserPreferenceService` which is called in `APP_INITIALIZE`, just the way we call `load()` for `UserInfoService`. That method will load the data and store the preferences in the future within the `UserPreferenceService` itself.

I don't think adding in `application-frame.component` will help in the future.

// @cseas @akashmane1598

Reply:
Not sure how `APP_INITIALIZE` handles ordering of calls. If we do that, then we need to make sure `APP_INITIALIZE` calls `UserInfoService` before `UserPreferenceService`, otherwise if it happens in the reverse order, then `UserInfoService` will return the default `ht-user`.

Maybe we can remove `APP_INITIALIZE` from `UserInfoService` altogether and move it to `UserPreferenceService`? But that would require me to modify files in Akash's PR.

Reply:
@jaywalker21 As assumed above, circular dependencies are not allowed in services. I tried adding `this.userPreferenceService.get('/v1/user/add');` to the `load()` method of `UserInfoService` and got this error.

<img width="450" alt="Screenshot 2022-08-10 at 7 31 24 PM" src="https://user-images.githubusercontent.com/29686866/183921317-3be77652-bfc8-4a96-94dd-976ef6ef068c.png">

Reply:
✅ Removed `APP_INITIALIZER` from `UserInfoService` and added it to `UserPreferenceService` instead which calls both `/user-info` and `/user/add` now.